base/shared/src/main/scala/scalaz/tc/joinsemilattice.scala

```diff
@@ -0,0 +1,59 @@
+package scalaz.tc
+
+import Predef._
+import scalaz.data.{ Const, Identity }
+
+/**
+ * In abstract algebra (and order theory), a join-semilattice is a partially ordered set S together with
+ * with an operation "join" which computes the supremum (least upper bound) for every pair of finite,
+ * nonempty subsets of S.
```

Comment:
Either "for every pair of elements", or "for a finite, nonempty subset", but not "for every pair of finite, non-empty subsets" 😃

Reply:
Good catches all around. I'll formalize what I mean